TreasureData/TreasureData.m

```diff
@@ -694,6 +697,68 @@ - (void)resetUniqId {
     [[NSUserDefaults standardUserDefaults] setObject:_UUID forKey:storageKeyOfUuid];
 }
 
+#pragma mark - Personalization API
+
+- (void)fetchUserSegments: (nonnull NSArray<NSString *> *)audienceTokens
+                     keys: (nonnull NSDictionary<NSString *, id> *)keys
+                  options: (nullable NSDictionary<TDRequestOptionsKey, id> *)options
+        completionHandler: (void (^_Nonnull)(NSArray* _Nullable jsonResponse, NSError* _Nullable error)) handler {
+    NSString *cdpEndpoint = self.cdpEndpoint ? self.cdpEndpoint : defaultCdpEndpoint;
+    NSString *audienceString = [NSString
+                                stringWithFormat:@"&token=%@",
+                                [audienceTokens componentsJoinedByString: @","]];
+    NSMutableString *keyString = [[NSMutableString alloc] initWithString: @""];
+    for (NSString *key in keys) {
+        [keyString appendFormat:@"&key.%@=%@", key, keys[key]];
```

Comment:
Should keys be URL encoded?

Reply:
Nice catch :) Yes it should, thank you very much. Fixed